sklearn/tests/test_base.py

```diff
@@ -314,48 +313,132 @@ def transform(self, X, y=None):
     assert_equal(e.scalar_param, cloned_e.scalar_param)
 
 
-class TreeNoVersion(DecisionTreeClassifier):
-    def __getstate__(self):
-        return self.__dict__
+def test_pickle_version_warning_is_not_raised_with_matching_version():
+    iris = datasets.load_iris()
+    tree = DecisionTreeClassifier().fit(iris.data, iris.target)
+    tree_pickle = pickle.dumps(tree)
+    assert_true(b"version" in tree_pickle)
+    tree_restored = assert_no_warnings(pickle.loads, tree_pickle)
+
+    # test that we can predict with the restored decision tree classifier
+    score_of_original = tree.score(iris.data, iris.target)
+    score_of_restored = tree_restored.score(iris.data, iris.target)
+    assert_equal(score_of_original, score_of_restored)
 
 
 class TreeBadVersion(DecisionTreeClassifier):
     def __getstate__(self):
         return dict(self.__dict__.items(), _sklearn_version="something")
 
 
-def test_pickle_version_warning():
-    # check that warnings are raised when unpickling in a different version
+pickle_error_message = (
+    "Trying to unpickle estimator {estimator} from "
+    "version {old_version} when using version "
+    "{current_version}. This might "
+    "lead to breaking code or invalid results. "
+    "Use at your own risk.")
 
-    # first, check no warning when in the same version:
-    iris = datasets.load_iris()
-    tree = DecisionTreeClassifier().fit(iris.data, iris.target)
-    tree_pickle = pickle.dumps(tree)
-    assert_true(b"version" in tree_pickle)
-    assert_no_warnings(pickle.loads, tree_pickle)
 
-    # check that warning is raised on different version
+def test_pickle_version_warning_is_issued_upon_different_version():
+    iris = datasets.load_iris()
     tree = TreeBadVersion().fit(iris.data, iris.target)
     tree_pickle_other = pickle.dumps(tree)
-    message = ("Trying to unpickle estimator TreeBadVersion from "
-               "version {0} when using version {1}. This might lead to "
-               "breaking code or invalid results. "
-               "Use at your own risk.".format("something",
-                                              sklearn.__version__))
+    message = pickle_error_message.format(estimator="TreeBadVersion",
+                                          old_version="something",
+                                          current_version=sklearn.__version__)
     assert_warns_message(UserWarning, message, pickle.loads, tree_pickle_other)
 
-    # check that not including any version also works:
+
+class TreeNoVersion(DecisionTreeClassifier):
```

Comment:
My 2c: moving things around like this adds unnecessary noise in the diff thus making it harder to review without any significant benefit.